File: src/test/java/org/ccci/gto/android/common/support/v4/util/MultiKeyLruCacheTest.java
package org.ccci.gto.android.common.support.v4.util;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertNotNull;
import static org.junit.Assert.assertNull;

import org.junit.Test;

public class MultiKeyLruCacheTest {
    @Test
    public void testNormalUsage() throws Exception {
        final MultiKeyLruCache<String, String> cache = new MultiKeyLruCache<>(1);
        final String sharedVal = "common value";

        // populate cache
        for (int i = 0; i < 100; i++) {
            cache.putMulti("KEY" + Integer.toString(i), sharedVal);
        }

        // test cache
        assertEquals(100, cache.size());
        for (int i = 0; i < 100; i++) {
            assertEquals(sharedVal, cache.get("KEY" + Integer.toString(i)));
        }

        // force eviction
        cache.putMulti("OTHERKEY", "new val");

        // test cache
        assertNotNull(cache.get("OTHERKEY"));
        assertEquals(1, cache.size());
        for (int i = 0; i < 100; i++) {
            assertNull(cache.get("KEY" + Integer.toString(i)));
        }
    }
}

File: gto-support-util/src/test/java/org/ccci/gto/android/common/util/MultiKeyLruCacheTest.kt
package org.ccci.gto.android.common.util

import androidx.test.ext.junit.runners.AndroidJUnit4
import kotlin.test.assertEquals
import kotlin.test.assertNotNull
import kotlin.test.assertNull
import org.junit.Test
import org.junit.runner.RunWith

@RunWith(AndroidJUnit4::class)
class MultiKeyLruCacheTest {
    @Test
    fun testNormalUsage() {
        val cache = MultiKeyLruCache<String, String>(1)
        val sharedVal = "common value"

        // populate cache
        for (i in 0..99) cache.putMulti("KEY$i", sharedVal)

        // test cache
        assertEquals(100, cache.size())
        for (i in 0..99) assertEquals(sharedVal, cache["KEY$i"])

        // force eviction
        cache.putMulti("OTHERKEY", "new val")

        // test cache
        assertNotNull(cache["OTHERKEY"])
        assertEquals(1, cache.size())
        for (i in 0..99) {
            assertNull(cache["KEY$i"])
        }
    }
}
